### PR TITLE
8304927: Update java/net/httpclient/BasicAuthTest.java to check basic auth over HTTP/2

### PR DESCRIPTION
**Pull request by dfuch**
The BasicAuthTest only tests basic authentication with clear HTTP/1.1. 
This changes upgrades it to also test against TLS,  as well as HTTP/2.
The bulk of the fix is to move the implementation of a HttpBasicAuthFilter from DigestEchoServer.java to HttpServerAdapters. This makes it possible to enable basic auth in the HTTP/2 test server.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304927](https://bugs.openjdk.org/browse/JDK-8304927): Update java/net/httpclient/BasicAuthTest.java to check basic auth over HTTP/2

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [446b1b48](https://git.openjdk.org/jdk/pull/13189/files/446b1b481e09754d03ae81fa9177ceb4d3d87f04)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13189/head:pull/13189` \
`$ git checkout pull/13189`

Update a local copy of the PR: \
`$ git checkout pull/13189` \
`$ git pull https://git.openjdk.org/jdk.git pull/13189/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13189`

View PR using the GUI difftool: \
`$ git pr show -t 13189`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13189.diff">https://git.openjdk.org/jdk/pull/13189.diff</a>

</details>

### CODE REVIEW

**Review by jaikiran — APPROVED** [review]
Looks good to me.

File: test/jdk/java/net/httpclient/BasicAuthTest.java [hunk 1]
```diff
@@ -95,16 +105,18 @@ public static void main(String[] args) throws Exception {
...
-            req = HttpRequest.newBuilder(uri)
-                             .POST(BodyPublishers.ofString(POST_BODY))
+            req = builder.copy().POST(BodyPublishers.ofString(POST_BODY))
                              .build();
+            System.out.println("\n\nSending POST request: " + req);
             resp = client.send(req, BodyHandlers.ofString());
             ok = resp.statusCode() == 200;
 
             if (!ok || ca.count != 1)
                 throw new RuntimeException("Test failed");
```
Comment (jaikiran) [inline test/jdk/java/net/httpclient/BasicAuthTest.java:115]: Hello Daniel, while we are changing this test, do you think we should also update this exception message, here and other places where we throw this, to also include the `ca.count` in the exception message to help debug any potential failures?

Comment (dfuch) [reply test/jdk/java/net/httpclient/BasicAuthTest.java]: Done

**Pull request closed without merging**